pandas/_libs/tslibs/conversion.pyx

```diff
@@ -524,10 +524,14 @@ cdef inline check_overflows(_TSObject obj):
     # GH#12677
     if obj.dts.year == 1677:
         if not (obj.value < 0):
-            raise OutOfBoundsDatetime
+            raise OutOfBoundsDatetime(
+                f'Timestamp cannot be converted within implementation bounds'
```

Comment:
for me, this explains what a OutOfBoundsDatetime exception is rather than the context of how it failed. can the message include the value and the value bound.

Reply:
Have implemented it let me know if there is a better way, not sure about the current one.